gemgis/vector.py

```diff
@@ -19,100 +20,523 @@
+    gdf['points'] = [list(i.coords) for i in gdf.geometry]
+    df = pd.DataFrame(gdf).explode('points')
+    df[['X', 'Y']] = pd.DataFrame(df['points'].tolist(), index=df.index)
+    if reset_index:
+        df = df.reset_index()
+    gdf = gpd.GeoDataFrame(df, geometry=gpd.points_from_xy(df.X, df.Y), crs=crs)
+
+    # Dropping id column
+    if 'id' in gdf and drop_id:
+        gdf = gdf.drop('id', axis=1)
+
+    # Dropping index column
+    if 'index' in gdf and drop_index:
+        gdf = gdf.drop('index', axis=1)
+
+    # Dropping points column
+    if 'points' in gdf and drop_points:
+        gdf = gdf.drop('points', axis=1)
+
+    # Limiting the extent of the data
+    if bbox is not None:
+        gdf = gdf[(gdf.X > bbox[0]) & (gdf.X < bbox[1]) & (gdf.Y > bbox[2]) & (gdf.Y < bbox[3])]
+
+    return gdf
+
+
+def extract_xy_points(gdf: gpd.geodataframe.GeoDataFrame,
+                      reset_index: bool = True,
+                      drop_id: bool = True,
+                      overwrite_xy: bool = False,
+                      target_crs: str = None,
+                      bbox: Optional[Sequence[float]] = None) -> gpd.geodataframe.GeoDataFrame:
+    """
+   Extracting x,y coordinates from a GeoDataFrame (Points) and returning a GeoDataFrame with x,y
+   coordinates as additional columns
+   Args:
+       gdf (gpd.geodataframe.GeoDataFrame): GeoDataFrame created from vector data containing elements of geom_type Point
+       reset_index (bool): Variable to reset the index of the resulting GeoDataFrame, default True
+       drop_id (bool): Variable to drop the id column, default True
+       overwrite_xy (bool): Variable to overwrite existing X and Y values, default False
+       target_crs (str, pyproj.crs.crs.CRS): Name of the CRS provided to reproject coordinates of the GeoDataFrame
+       bbox (list): Values (minx, maxx, miny, maxy) to limit the extent of the data
+   Return:
+       gdf (gpd.geodataframe.GeoDataFrame): GeoDataFrame with appended x,y columns and optional columns
+   """
+
+    # Checking that gdf is of type GepDataFrame
+    if not isinstance(gdf, gpd.geodataframe.GeoDataFrame):
+        raise TypeError('Loaded object is not a GeoDataFrame')
+
+    # Check that all entries of the gdf are of type Point
+    if not all(gdf.geom_type == 'Point'):
+        raise TypeError('All GeoDataFrame entries must be of geom_type Point')
+
+    # Checking that the bbox is of type None or list
+    if not isinstance(bbox, (type(None), list)):
+        raise TypeError('The bbox values must be provided as list')
+
+    # Checking that the bbox list only has four elements
+    if isinstance(bbox, list) and not len(bbox) == 4:
+        raise ValueError('Provide minx, maxx, miny and maxy values for the bbox')
+
+    # Checking that all elements of the list are of type int or float
+    if isinstance(bbox, list) and not all(isinstance(i, (int, float)) for i in bbox):
+        raise TypeError('Bbox values must be of type float or int')
```

Comment:
```suggestion
    if bbox is not None:
        if not isinstance(bbox, Sequence):
            raise TypeError('The bbox values must be provided as a sequence')

        # Checking that the bbox list only has four elements
        if len(bbox) != 4:
            raise ValueError('Provide minx, maxx, miny and maxy values for the bbox')

        # Checking that all elements of the list are of type int or float
        if not all(isinstance(bound, (int, float)) for bound in bbox):
            raise TypeError('Bbox values must be of type float or int')
```

Reply:
Forgot to check for variable names as well. We shouldn't use i as a variable if it doesn't mean an index over something.